src/errorpage.cc

```diff
@@ -810,67 +811,59 @@ ErrorState::~ErrorState()
 int
 ErrorState::Dump(MemBuf * mb)
 {
-    MemBuf str;
-    char ntoabuf[MAX_IPSTRLEN];
+    PackableStream out(*mb);
+    SBufStream body;
+    const auto &encoding = CharacterSet::RFC3986_UNRESERVED();
+
+    body << "CacheHost: " << getMyHostname() << "\r\n" <<
+        "ErrPage: " << errorPageName(type) << "\r\n";
+    if (xerrno)
+        body << "Err: (" << xerrno << ") " << strerror(xerrno) << "\r\n";
+    else
+        body << "Err: [none]\r\n";
 
-    str.reset();
-    /* email subject line */
-    str.appendf("CacheErrorInfo - %s", errorPageName(type));
-    mb->appendf("?subject=%s", rfc1738_escape_part(str.buf));
-    str.reset();
-    /* email body */
-    str.appendf("CacheHost: %s\r\n", getMyHostname());
-    /* - Err Msgs */
-    str.appendf("ErrPage: %s\r\n", errorPageName(type));
-
-    if (xerrno) {
-        str.appendf("Err: (%d) %s\r\n", xerrno, strerror(xerrno));
-    } else {
-        str.append("Err: [none]\r\n", 13);
-    }
 #if USE_AUTH
     if (auth_user_request.getRaw() && auth_user_request->denyMessage())
-        str.appendf("Auth ErrMsg: %s\r\n", auth_user_request->denyMessage());
+        body << "Auth ErrMsg: " << auth_user_request->denyMessage() << "\r\n";
 #endif
+
     if (dnsError)
-        str.appendf("DNS ErrMsg: " SQUIDSBUFPH "\r\n", SQUIDSBUFPRINT(*dnsError));
+        body << "DNS ErrMsg: " << *dnsError << "\r\n";
 
-    /* - TimeStamp */
-    str.appendf("TimeStamp: %s\r\n\r\n", Time::FormatRfc1123(squid_curtime));
+    body << "TimeStamp: " << Time::FormatRfc1123(squid_curtime) << "\r\n\r\n"
+         "ClientIP: " << src_addr << "\r\n";
```

Comment:
This new/PR code is implicitly using Ip::Address::toUrl(). Official code uses Ip::Address::toStr(). The two methods produce different output. If this change is desirable, please disclose it (with its rationale and effect) in the PR description. Otherwise, refactor to avoid an undesirable change.

Reply:
Thanks for picking this up; reverted to the previous output

Reply:
This change is still present. Only change was the line it occurs on - which confuses github into marking it "outdated".